src/libstd/sys/cloudabi/condvar.rs

```diff
@@ -131,21 +136,27 @@ impl Condvar {
                 ..mem::zeroed()
             },
         ];
-        let mut events: [abi::event; 2] = mem::uninitialized();
-        let mut nevents: usize = mem::uninitialized();
-        let ret = abi::poll(subscriptions.as_ptr(), events.as_mut_ptr(), 2, &mut nevents);
+        let mut events: [mem::MaybeUninit<abi::event>; 2] = [mem::MaybeUninit::uninit(); 2];
+        let mut nevents: mem::MaybeUninit<usize> = mem::MaybeUninit::uninit();
+        let ret = abi::poll(
+            subscriptions.as_ptr(),
+            mem::MaybeUninit::first_ptr_mut(&mut events),
+            2,
+            nevents.get_mut()
```

Comment:
Does this *have to* be a reference (as opposed to a raw pointer)?

Reply:
My only concern was not wanting to change the CloudABI::abi definitions, but it looks like it just passes the reference to a syscall as a raw pointer anyway so it should be fine. The changes are in a new commit I just pushed.